packages/teleport/src/components/TextSelectCopy/TextSelectCopyMulti.test.tsx

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright 2022 Gravitational, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+
+import { render, screen, fireEvent, waitFor } from 'design/utils/testing';
+
+import { TextSelectCopyMulti } from './TextSelectCopyMulti';
+
+const originalNavigator = window.navigator;
+
+beforeEach(() => {
+  jest.useFakeTimers();
+  Object.assign(window.navigator, {
+    clipboard: {
+      writeText: jest.fn().mockImplementation(() => Promise.resolve()),
+    },
+  });
+});
+
+afterEach(() => {
+  jest.resetAllMocks();
+  Object.assign(window.navigator, originalNavigator);
```

Comment:
This won't do anything since `window.navigator` and `originalNavigator` points to the same object reference.
```
const originalNavigator = window.navigator; // this will alias window.navigator not clone it
```
In result `window.navigator` will have mocked method even after test is finished

Reply:
ah you are correct. though i was also thinking the resetting is unnecessary because if i am not mistaken, each test file is sandboxed and will not affect other test files. so I think it's fine to not handle resetting it.

also the `spyOn` does not work because the clipboard is not defined? there is a library that handles this stubbing for us: https://github.com/testing-library/user-event, though atm it feels like an overkill, let me know what you think

Reply:
user-event is an official testing library extension. It provides API that better emulates the user behavior. I think we could use it anyway.

The alternative would be to add `setupTests.js` file:
with
```
Object.assign(navigator, {
  clipboard: {
    writeText: () => {},
  },
});
```
+ add `setupFilesAfterEnv: ['<rootDir>/setupTests.js'],` to `jest.config.js`